packages/cli/src/chat/chat-service.ts

```diff
@@ -245,21 +264,39 @@ export class ChatService {
 		return execution;
 	}
 
-	private processIncomingData(data: RawData) {
+	private stringifyRawData(data: RawData) {
 		const buffer = Array.isArray(data)
 			? Buffer.concat(data.map((chunk) => Buffer.from(chunk)))
 			: Buffer.from(data);
 
-		const message = jsonParse<ChatMessage>(buffer.toString('utf8'));
+		return buffer.toString('utf8');
+	}
 
-		if (message.files) {
-			message.files = message.files.map((file) => ({
-				...file,
-				data: file.data.includes('base64,') ? file.data.split('base64,')[1] : file.data,
-			}));
-		}
+	private cleanupSession(session: Session) {
+		session.connection.terminate();
+		clearInterval(session.intervalId);
+	}
+
+	private prepareChatMessage(message: string): ChatMessage {
+		try {
+			const parsedMessage = chatMessageSchema.parse(jsonParse(message));
 
-		return message;
+			if (parsedMessage.files) {
+				parsedMessage.files = parsedMessage.files.map((file) => ({
+					...file,
+					data: file.data.includes('base64,') ? file.data.split('base64,')[1] : file.data,
+				}));
+			}
+
+			return parsedMessage;
+		} catch (error) {
+			if (error instanceof z.ZodError) {
+				throw new Error(
```

Comment:
I assume this should never happen as we control both client and server? We should use `UnexpectedError` in that case